cli/cmd/inject_test.go

```diff
@@ -61,89 +59,77 @@ func testUninjectAndInject(t *testing.T, tc testCase) {
-	emptyProxyVersionConfig := testInstallConfig(ctx)
-	emptyProxyVersionConfig.Global.Version = "test-inject-control-plane-version"
-	emptyProxyVersionConfig.Proxy.ProxyVersion = ""
-
-	overrideConfig := testInstallConfig(ctx)
-	overrideConfig.Proxy.ProxyVersion = "override"
+	defaultValues, err := testInstallValues()
+	if err != nil {
+		log.Fatalf("Unexpected error: %v", err)
+	}
+	defaultValues.Global.LinkerdVersion = "test-inject-control-plane-version"
+	defaultValues.Global.Proxy.Image.Version = "test-inject-proxy-version"
+	defaultValues.DebugContainer.Image.Version = "test-inject-debug-version"
 
-	proxyResourceConfig := testInstallConfig(ctx)
-	proxyResourceConfig.Proxy.ProxyVersion = defaultConfig.Proxy.ProxyVersion
-	proxyResourceConfig.Proxy.Resource = &config.ResourceRequirements{
-		RequestCpu:    "110m",
-		RequestMemory: "100Mi",
-		LimitCpu:      "160m",
-		LimitMemory:   "150Mi",
+	overrideConfig, err := testInstallValues()
+	if err != nil {
+		log.Fatalf("Unexpected error: %v", err)
 	}
+	overrideConfig.Global.Proxy.Image.Version = "override"
 
-	cniEnabledConfig := testInstallConfig(ctx)
-	cniEnabledConfig.Proxy.ProxyVersion = defaultConfig.Proxy.ProxyVersion
-	cniEnabledConfig.Global.CniEnabled = true
+	proxyResourceConfig, err := testInstallValues()
+	if err != nil {
+		log.Fatalf("Unexpected error: %v", err)
+	}
+	proxyResourceConfig.Global.Proxy.Image.Version = defaultValues.Global.Proxy.Image.Version
+	proxyResourceConfig.Global.Proxy.Resources = &linkerd2.Resources{
+		CPU: linkerd2.Constraints{
+			Request: "110m",
+			Limit:   "160m",
+		},
+		Memory: linkerd2.Constraints{
+			Request: "100Mi",
+			Limit:   "150Mi",
+		},
+	}
 
-	proxyIgnorePortsOptions, err := testInstallOptions()
+	cniEnabledConfig, err := testInstallValues()
 	if err != nil {
 		log.Fatalf("Unexpected error: %v", err)
 	}
-	proxyIgnorePortsOptions.ignoreInboundPorts = []string{"22", "8100-8102"}
-	proxyIgnorePortsOptions.ignoreOutboundPorts = []string{"5432"}
-	_, proxyIgnorePortsConfig, err := proxyIgnorePortsOptions.validateAndBuild(ctx, "", nil)
+	cniEnabledConfig.Global.Proxy.Image.Version = defaultValues.Global.Proxy.Image.Version
+	cniEnabledConfig.Global.CNIEnabled = true
+
+	proxyIgnorePortsConfig, err := testInstallValues()
 	if err != nil {
-		log.Fatalf("test install proxy-ignore options must be valid: %s", err)
+		log.Fatalf("Unexpected error: %v", err)
 	}
+	proxyIgnorePortsConfig.Global.ProxyInit.IgnoreInboundPorts = "22,8100-8102"
+	proxyIgnorePortsConfig.Global.ProxyInit.IgnoreOutboundPorts = "5432"
 
 	testCases := []testCase{
 		{
 			inputFileName:    "inject_emojivoto_deployment.input.yml",
 			goldenFileName:   "inject_emojivoto_deployment.golden.yml",
 			reportFileName:   "inject_emojivoto_deployment.report",
 			injectProxy:      true,
-			testInjectConfig: defaultConfig,
-		},
-		{
-			inputFileName:    "inject_emojivoto_deployment.input.yml",
-			goldenFileName:   "inject_emojivoto_deployment_empty_version_config.golden.yml",
```

Comment:
These tests with empty versions were deleted as `Values` is our final config,(on which updates through flags, annotations were overridden before rendering) and hence having a test with their versions field as empty does not make sense